### DISCUSSION TITLE
How to dynamically generate sitemap.xml/robots.txt for project using host rewrites?

### DISCUSSION THREAD

**Discussion by TrustyTechSG** [vercel/next.js#57545] [App Router]
Hi we are using rewrites to dynamically generate site base on visiting hostname, which works great. but for the sitemap.xml/robots.txt, as both of them need to place direct under root folder of app directory, then how can we generate them base on the hostname?

```
  async rewrites() {
    return {
      beforeFiles: [
        {
          has: [
            {
              type: "host",
              value: "(?<host>.*)",
            },
          ],
          source: "/:path*",
          destination: "/hosts/:host/:path*",
        },
```

**Comment by vignesh-gupta** [comment]
Hey @TrustyTechSG, If you are looking for dynamically create `sitemap` and `robots` then you have to create `sitemap.ts` and `robots.ts` in the app dir. 

Details and syntax can be found here - [Generate a Sitemap](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/sitemap#generate-a-sitemap) & [Generate a Robots file](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/robots#generate-a-robots-file)

**Reply by TrustyTechSG** [reply]
Hi there. thank you for the reply, yes I saw this, but these function are not passing in any params is it? We need the params to get the hostname.

We have rewrite rules to rewrite all visit under
`/hosts/[hostname]`
So we need to put the sitemap.ts and robots.ts under /hosts/[hostname] folder and get the hostname params as argument, it is currently not possible is it?
